server/src/main/java/io/druid/segment/realtime/appenderator/Appenderator.java

```diff
@@ -72,8 +73,8 @@ default AppenderatorAddResult add(SegmentIdentifier identifier, InputRow row, Su
    * Committer is guaranteed to be *created* synchronously with the call to add, but will actually be used
    * asynchronously.
    * <p>
-   * The add, clear, persist, persistAll, and push methods should all be called from the same thread to keep the
-   * metadata committed by Committer in sync.
+   * If committer is not provided, any data are NOT persisted. If it's provided, the add, clear, persist, persistAll,
```

Comment:
Sorry, the Javadoc was wrong. It should be "no metadata is persisted". Fixed it now.

`Committer` is about persisting intermediate metadata while `allowIncrementalPersists` is about persisting data ingested so far. So, as you said, if `allowIncrementalPersists` is set to false then committer is ignored. But if `allowIncrementalPersists`, `committer` can still be null to avoid persisting metadata.

Reply:
ah that makes sense now.
so, null committer  simply means commit metadata is not persisted.